src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import duke.task.*;
+import duke.ui.ErrorUi;
+import duke.ui.Ui;
+
+import java.io.IOException;
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private ErrorUi errorUi;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        errorUi = new ErrorUi();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException | IOException e) {
+            errorUi.printErrorMessage(ErrorUi.ERROR_DATA_LOAD);
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        start();
+        runCommandLoopUntilByeCommand();
+        exit();
+    }
+
+    public static void main(String[] args) {
+        new Duke("data/tasks.txt").run();
```

Comment:
Perhaps you could replace with magic literal with a constant? It would make it easier to locate and make changes to the file path for future use